pkg/controller/storage/bucket/aws_handler.go

```diff
@@ -63,6 +64,7 @@ func (h *S3BucketHandler) newS3Bucket(class *corev1alpha1.ResourceClass, instanc
 		},
 		Spec: *bucketSpec,
 	}
+	bucket.Status.SetUnbound()
```

Comment:
It would be nice if the zero value of things like this were a sane default (Unbound in this case). I believe we could do that by, for example:

```go
type BindingState int

const (
	BindingStateUnbound BindingState = iota
	BindingStateBound
)

func (s BindingState) MarshalJSON() ([]byte, error) {
  // Return a human friendly string name here
}
```

Reply:
The zero-value having a default of Unbound sounds nice to me, because then it's not an explicit action to set the status to unbound at the beginning every time.  It's only an explicit action later on to set it to bound when that happens.  I would need to play with this and test it out a bit to make sure that the binding state gets committed to the k8s API server correctly, and when a `kubectl get` is done then the binding state is shown appropriately as well.  I believe you are right that implementing the `Marshaler` and `Unmarshaler` interfaces would do the trick.

Reply:
I decided to track this idea in #320 so it can be followed up on in a future PR.  I wasn't comfortable with the churn and scope that the PR was taking on with including that change.  Doing that in a new PR in isolation makes sense to me.